contracts/token/ERC20/ERC20.sol

```diff
@@ -241,25 +241,20 @@ contract ERC20 is Context, IERC20, IERC20Metadata {
     ) internal virtual {
         if (from == address(0)) {
             _totalSupply += amount;
-            unchecked {
-                // Overflow not possible: balance + amount is at most totalSupply + amount, which is checked above.
-                _balances[to] += amount;
-            }
-        } else if (to == address(0)) {
+        } else {
             uint256 fromBalance = _balances[from];
-            require(fromBalance >= amount, "ERC20: burn amount exceeds balance");
-            _totalSupply -= amount;
+            require(fromBalance >= amount, "ERC20: transfer amount exceeds balance");
             unchecked {
                 // Overflow not possible: amount <= fromBalance <= totalSupply.
                 _balances[from] = fromBalance - amount;
             }
+        }
+
+        if (to == address(0)) {
+            _totalSupply -= amount;
         } else {
-            uint256 fromBalance = _balances[from];
-            require(fromBalance >= amount, "ERC20: transfer amount exceeds balance");
             unchecked {
-                _balances[from] = fromBalance - amount;
-                // Overflow not possible: the sum of all balances is capped by totalSupply, and the sum is preserved by
-                // decrementing then incrementing.
+                // Overflow not possible: balance + amount is at most totalSupply + amount, which is checked above.
```

Comment:
That comment is wrong, totalSupply+amount is not checked.

A better comment would be balance+amount is at most totalSupply, which we know fits into a uint256